frontend/src/features/settings/hooks/creditAccount.ts

```diff
@@ -0,0 +1,254 @@
+import { getCompanyDataBySearch } from "../../idir/search/api/idirSearch";
+import { useNavigate } from "react-router-dom";
+import { ERROR_ROUTES } from "../../../routes/constants";
+import { SnackBarContext } from "../../../App";
+import { useContext } from "react";
+import {
+  CreditAccountStatusType,
+  UPDATE_STATUS_ACTIONS,
+  UpdateStatusData,
+} from "../types/creditAccount";
+import { getCompanyIdFromSession } from "../../../common/apiManager/httpRequestHandler";
+import { SnackbarAlertType } from "../../../common/components/snackbar/CustomSnackBar";
+import { CompanyProfile } from "../../manageProfile/types/manageProfile";
+
+/**
+ * Hook to fetch the company credit account details for the active user.
+ * @returns Query result of the company credit account details
+ */
+export const useGetCreditAccountQuery = () => {
+  return useQuery({
+    queryKey: ["credit-account", { companyId: getCompanyIdFromSession() }],
+    queryFn: getCreditAccount,
+  });
+};
+
+/**
+ * Hook to fetch the associated credit account details for the given company.
+ * @returns Query result of the company associated credit account details
+ */
+export const useGetCompanyCreditAccountQuery = (companyId: number) => {
+  return useQuery({
+    queryKey: ["credit-account", { companyId }],
+    queryFn: () => getCompanyCreditAccount(companyId),
+  });
+};
+
+/**
+ * Hook to fetch the company credit account details.
+ * @returns Query result of the company credit account details
+ */
+export const useGetCreditAccountUsersQuery = (creditAccountId: number) => {
+  return useQuery({
+    queryKey: [
+      "credit-account",
+      { companyId: getCompanyIdFromSession() },
+      "users",
+    ],
+    queryFn: () => getCreditAccountUsers(creditAccountId),
+  });
+};
+
+/**
+ * Hook to fetch the user information.
+ * @param clientNumber Id of the user to get information for
+ * @returns Query result of the user information
+ */
+export const useGetCompanyQuery = (clientNumber: string) => {
+  return useQuery({
+    queryKey: ["company-information", { clientNumber }],
+    queryFn: () =>
+      getCompanyDataBySearch(
+        {
+          searchEntity: "companies",
+          searchByFilter: "onRouteBCClientNumber",
+          searchString: clientNumber,
+        },
+        {
+          page: 0,
+          take: 1,
+        },
+      ),
+    enabled: clientNumber !== "",
+    staleTime: 0,
+  });
+};
+
+/**
+ * Hook to create a credit account.
+ * @returns Result of the create credit account action
+ */
+export const useCreateCreditAccountMutation = () => {
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const { setSnackBar } = useContext(SnackBarContext);
+
+  return useMutation({
+    mutationFn: createCreditAccount,
+    onSuccess: (response) => {
+      queryClient.setQueryData(
+        ["credit-account", { companyId: getCompanyIdFromSession() }],
+        response.data,
+      );
+      setSnackBar({
```

Comment:
Performing such tasks in the onSuccess or onError callbacks is possible for now, but I'm not sure about the future if react-query decides to remove these callbacks (like they did for useQuery for the previous version). Just keep this in mind.